### PR TITLE
Jwoo/ios custom parser for ios

### PR DESCRIPTION
**Pull request by nesalang**

### CODE REVIEW

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOBaseCardElement.h [hunk 1]
```diff
@@ -41,3 +43,10 @@ typedef NS_ENUM(NSInteger, ACRContainerStyle) {
 @property ACRCardElementType type;
 
 @end
+
+@protocol ACOIBaseCardElementParser
+
+@optional 
+- (UIView *)DeserializeToCustomUIElement:(NSData* )json;
```
Comment (bviglietta) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOBaseCardElement.h]: >DeserializeToCustomUIElement [](start = 12, length = 28)

Lowercase: deserializeToCustomUIElement

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRegistration.h [hunk 2]
```diff
@@ -21,4 +21,7 @@
 - (void) setActionRenderer:(ACRBaseActionElementRenderer *)renderer cardElementType:(NSNumber *)cardElementType;
 
 - (void) setBaseCardElementRenderer:(ACRBaseCardElementRenderer *)renderer cardElementType:(ACRCardElementType)cardElementType;
+
+- (void) setCustomElementRenderer:(NSObject<ACOIBaseCardElementParser> *)customElementParser;
```
Comment (bviglietta) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRRegistration.h]: >Renderer [](start = 25, length = 8)

I think this is setCustomElementParser, not setCustomElementRenderer.

File: source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModel/AdaptiveCardsSharedModel.vcxproj [hunk 3]
```diff
@@ -121,6 +121,7 @@
     <ClCompile Include="..\..\ObjectModel\Column.cpp" />
     <ClCompile Include="..\..\ObjectModel\ColumnSet.cpp" />
     <ClCompile Include="..\..\ObjectModel\Container.cpp" />
+    <ClCompile Include="..\..\ObjectModel\CustomObjectDelegateForIOS.cpp" />
```
Comment (bviglietta) [inline source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModel/AdaptiveCardsSharedModel.vcxproj]: ><ClCompile Include="..\..\ObjectModel\CustomObjectDelegateForIOS.cpp" /> [](start = 4, length = 72)

Is this a merge error or something? This PR doesn't add this file.

Comment (nesalang) [reply source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModel/AdaptiveCardsSharedModel.vcxproj]: thanks, it must be.

---
In reply to: [172385623](https://github.com/Microsoft/AdaptiveCards/pull/1210#discussion_r172385623) [](ancestors = 172385623)

**Pull request merged**